pkg/apis/pipeline/v1beta1/pipelinerun_validation.go

```diff
@@ -82,6 +83,24 @@ func (ps *PipelineRunSpec) Validate(ctx context.Context) (errs *apis.FieldError)
 		}
 	}
 
+	if ps.TasksTimeout != nil {
+		// tasksTimeout should be a valid duration of at least 0.
+		if ps.TasksTimeout.Duration < 0 {
+			errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("%s should be >= 0", ps.TasksTimeout.Duration.String()), "tasksTimeout"))
+		}
+
+		if ps.Timeout != nil {
+			if ps.TasksTimeout.Duration > ps.Timeout.Duration {
+				errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("%s should be <= Timeout duration", ps.TasksTimeout.Duration.String()), "tasksTimeout"))
```

Comment:
suggest lowercase first letter of `timeout` here since it's referring to the user-facing yaml/json field.